snow/validators/logger.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package validators
+
+import (
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"go.uber.org/zap"
+)
+
+var _ NodeStakeLogger = (*logger)(nil)
+
+// NodeStakeLogger allows logging stake changes of a specified validator
+type NodeStakeLogger SetCallbackListener
+
+type logger struct {
+	subnetID ids.ID
+	nodeID   ids.NodeID
+	log      logging.Logger
+}
+
+func NewLogger(subnetID ids.ID, targetNode ids.NodeID, log logging.Logger) NodeStakeLogger {
+	return &logger{
+		subnetID: subnetID,
+		nodeID:   targetNode,
+		log:      log,
+	}
+}
+
+func (l *logger) OnValidatorAdded(valID ids.NodeID, _ *bls.PublicKey, txID ids.ID, weight uint64) {
+	if valID == l.nodeID {
+		l.log.Info("tracking validator: validator added to validators set",
```

Comment:
I'm wondering if it would be a better UX to somehow describe that this is the _local_ node. As the code is written it doesn't really make a ton of sense... But maybe rather than taking in a single `targetNode` we could take in a `Set[ids.ID]` along with a `reason` that we could then include in the log.

Reply:
I thought about this idea too but preferred to code just what @jpop32  asked. I understand the ask was for node admins to be able to track some of *this node* relevant facts, rather than to be able to explore facts about other validators. @jpop32 what do you think?